model/operations.py

```diff
@@ -175,6 +176,24 @@ def knockout_undo(model, to_undo):
             model.reactions.get_by_id(r_id).knock_out()
     return removed
 
+def changebounds_undo(model, to_undo):
+    for rn in to_undo:
+        if model.reactions.has_id(rn['id']):
+            model.reactions.get_by_id(rn['id']).bounds = \
+                rn['lower_bound'], rn['upper_bound']
+    return {reaction['id'] for reaction in to_undo}
+
+
+async def changebounds_apply(model, to_apply):
```

Comment:
it is now possible to get rid of "knockout" concept altogether on the backend side, since knockout is just a corner case of changing bounds with the bounds being equal `[0, 0]`
I think the API should be refactored this way. It would require making changes in `iloop-to-model` and frontend as well.